src/globus_sdk/services/transfer/client.py

```diff
@@ -739,10 +743,11 @@ def add_endpoint_role(
         path = self.qjoin_path("endpoint", endpoint_id_s, "role")
         return self.post(path, data=role_data)
 
+    # REVIEW: why do we call str() on server ID above but not on role_id here?
```

Comment:
I'm pretty sure we do not need to do so. I would guess that the source of this inconsistency is that `server_id` values are integers, but `role_id` values are UUIDs.

Perhaps the right solution is `server_id: Union[str, int]`, `role_id: ID_PARAM_TYPE`, and `safe_stringify()` or `str()` on everything.